### PR TITLE
Spring boot online book store add security

### PR DESCRIPTION
**Pull request by snowfallen**

### CODE REVIEW

File: src/main/java/book/store/model/User.java [hunk 1]
```diff
@@ -2,18 +2,29 @@
...
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Entity
 @Getter
 @Setter
 @Table(name = "users")
-public class User {
+public class User implements UserDetails {
+    private static final String ROLE_PREFIX = "ROLE_";
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/model/User.java]: not needed

File: src/main/resources/db/changelog/changes/06-fill-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+  - changeSet:
+      id: fill-roles-table
+      author: snowfall
+      changes:
+        - insert:
+            tableName: roles
+            columns:
+              - column: {name: "name", value: "ADMIN"}
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/changes/06-fill-roles-table.yaml]:
```suggestion
              - column: {name: "name", value: "ROLE_ADMIN"}
```

File: src/main/java/book/store/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    private Long id;
+    @Enumerated(EnumType.STRING)
+    @Column(name = "name", nullable = false, unique = true)
+    private RoleName name;
+    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT")
+    private boolean isDeleted = false;
+
+    public enum RoleName {
+        ADMIN,
+        USER
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/model/Role.java]:
```suggestion
        ROLE_ADMIN,
        ROLE_USER
```

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]
LGTM!

**Pull request merged**